frontend/src/app/organization/organization-contact/organization-contact.component.ts

```diff
@@ -1,46 +1,72 @@
-import {Observable} from 'rxjs/Observable';
-import {User} from '../../user';
-import {UserService} from '../../user.service';
-
+import { AuthService } from '../../auth.service';
+import { User } from '../../user';
+import { UserService } from '../../user.service';
 
 @Component({
   selector: 'volontulo-organization-contact',
   templateUrl: './organization-contact.component.html',
   styleUrls: ['./organization-contact.component.scss']
 })
-export class OrganizationContactComponent implements OnChanges {
-  @ViewChild('contactForm') contactForm: NgForm;
-  @Output() contact = new EventEmitter<OrganizationContactPayload>();
+
+export class OrganizationContactComponent implements OnInit, OnChanges {
   @Input() contactStatus: ContactStatus;
+  @Output() contact = new EventEmitter<OrganizationContactPayload>();
   submitDisabled = false;
-  alertSuccessClosed = true;
-  alertErrorClosed = true;
   user$: Observable<User> = this.authService.user$;
   getFullName = this.userService.getFullName;
 
-  constructor(private authService: AuthService, private userService: UserService) {
+  public fg: FormGroup = this.fb.group({
+    name: ['', [Validators.required, Validators.minLength(3), Validators.maxLength(30)]],
+    email: ['', [Validators.required, Validators.minLength(3), Validators.maxLength(30), Validators.email]],
+    phone_no: ['', [Validators.required, Validators.minLength(9), Validators.maxLength(9), Validators.pattern(/^[0-9]{9}$/)]],
+    message: ['', [Validators.required, Validators.minLength(10), Validators.maxLength(2000)]],
+    honeyBunny: ['']
+  });
+  public success: null | boolean = null;
+
+
+  constructor(
+    private fb: FormBuilder,
+    private authService: AuthService,
+    private userService: UserService
+  ) { }
+
+  ngOnInit() {
+    this.user$
+    .pipe(
+      filter(user => user !== null),
+      take(1),
+    )
+    .subscribe(user => {
+      this.fg.controls.name.setValue(this.getFullName(user));
+      this.fg.controls.email.setValue(user.email);
+      this.fg.controls.phone_no.setValue(user.phoneNo);
+      });
   }
 
   ngOnChanges(changes: SimpleChanges) {
     if (changes.contactStatus.currentValue && changes.contactStatus.currentValue.status === 'success') {
-      this.alertSuccessClosed = false;
-      this.contactForm.reset();
+      this.success = true;
+      this.fg.controls.message.reset();
     } else if (changes.contactStatus.currentValue && changes.contactStatus.currentValue.status === 'error') {
-      this.alertErrorClosed = false;
+      this.success = false;
     }
     this.submitDisabled = false;
   }
 
   onSubmit() {
-    if (!this.contactForm.value.honeyBunny) {
-      this.contact.emit(this.contactForm.value as OrganizationContactPayload);
+    if (this.fg.valid && !this.fg.value.honeyBunny) {
       this.submitDisabled = true;
+      delete this.fg.value.honeyBunny;
+      this.contact.emit(this.fg.value as OrganizationContactPayload);
```

Comment:
What is `this.contact` event emitter for?

Reply:
I didn't change component structure, and how it handles sending value from contact form. If I understood it correctly, form is send not from child component directly, but from parent component:
organization.component.ts:
```
  onContact(organizationContact: OrganizationContactPayload) {
    this.organization$.pipe(take(1))
      .subscribe(org => this.organizationService.sendContactForm(org, organizationContact));
  }
```
organization.component.html:
`<volontulo-organization-contact
  *ngIf="!(isUserOrgMember$ | async)"
  (contact)="onContact($event)"
  [contactStatus]="contactStatus$ | async">
</volontulo-organization-contact>`

Reply:
Sorry - I didn't note it was moved not added 👍